### PR TITLE
Authenticate contributions

### PR DESCRIPTION
**Pull request by davidfurey**
## Why are you doing this?

Users need to be logged in to become monthly contributors.  This also fetches their email preferences from identity.

### CODE REVIEW

File: app/lib/net/IPv4.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+object IPv4 {
+
+  val netmaskClassA = 0xff000000
+  val netmaskClassB = 0xfff00000
+  val netmaskClassC = 0xffff0000
+
+  val privatePrefixClassA = 10 << 24
+  val privatePrefixClassB = 172 << 24 | 16 << 16
+  val privatePrefixClassC = 192 << 24 | 168 << 16
+
```
Comment (rtyley) [inline app/lib/net/IPv4.scala]: This is nice code, I'm just wondering if maybe we should use [`Inet4Address.isSiteLocalAddress()`](https://docs.oracle.com/javase/8/docs/api/java/net/Inet4Address.html#isSiteLocalAddress--) instead? The `Inet4Address` class already has parsing code, and `isSiteLocalAddress()` [does the same job](https://stackoverflow.com/a/5619409/438886).

https://stackoverflow.com/a/5619409/438886

Comment (davidfurey) [inline app/lib/net/IPv4.scala]: I thought this was an improvement on the regex that frontend used, but isSiteLocalAddress is even better.  Thanks!

Comment (davidfurey) [inline app/lib/net/IPv4.scala]: I don't think that Inet4Address has the parsing code?

Comment (rtyley) [inline app/lib/net/IPv4.scala]: You can use Guava! https://google.github.io/guava/releases/20.0/api/docs/com/google/common/net/InetAddresses.html#forString-java.lang.String-

**Pull request merged**